lib/pbench/server/api/resources/datasets_metadata.py

```diff
@@ -77,9 +77,20 @@ def _get(self, json_data: JSON, request: Request) -> Response:
         name = json.get("name")
         keys = json.get("metadata")
 
-        self.logger.info("GET metadata {} for {}", keys, name)
         try:
-            metadata = self._get_metadata(name, keys)
+            dataset = Dataset.query(name=name)
+        except DatasetError as e:
+            self.logger.warning("Dataset {!r} not found: {}", name, str(e))
+            raise APIAbort(HTTPStatus.BAD_REQUEST, f"Dataset {name!r} not found")
+
+        # Validate the authenticated user's authorization for the combination
+        # of "owner" and "access".
+        self._check_authorization(
+            str(dataset.owner_id), dataset.access, check_role=API_OPERATION.READ
+        )
+
+        try:
+            metadata = self._get_dataset_metadata(dataset, keys)
         except DatasetNotFound:
             raise APIAbort(HTTPStatus.BAD_REQUEST, f"Dataset {name} not found")
```

Comment:
It would be nice if this message were (more) distinct from the message produced at line 84.  (Given that we should have aborted at line 84 if the dataset were not found, should this one be an internal error?

Reply:
That's a "cut and paste" error, more or less. The previous version called the "wrapper" metadata function which looked up the dataset. Now that it doesn't, I should probably just let this exception fly. OK, one more revision...